presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java

```diff
@@ -96,20 +204,37 @@ public static Slice regexpExtractAll(@SqlType(StandardTypes.VARCHAR) Slice sourc
     @Description("group(s) extracted using the given pattern")
     @ScalarFunction
     @SqlType("array<varchar>")
-    public static Slice regexpExtractAll(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(RegexpType.NAME) Pattern pattern, @SqlType(StandardTypes.BIGINT) long group)
+    public static Slice regexpExtractAll(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(RegexpType.NAME) Regex pattern, @SqlType(StandardTypes.BIGINT) long group)
     {
-        Matcher matcher = pattern.matcher(source.toString(UTF_8));
-        validateGroup(group, matcher);
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), 1024);
-        while (matcher.find()) {
-            String string = matcher.group(Ints.checkedCast(group));
-            if (string == null) {
+        Matcher matcher = pattern.matcher(source.getBytes());
+        validateGroup(group, matcher.getEagerRegion());
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 32);
+        int iGroup = Ints.checkedCast(group);
```

Comment:
This variable name is weird. I'd call it `group` and then rename `group` to `inputGroup`